tutorials/intermediate/blitting.py

```diff
@@ -0,0 +1,183 @@
+"""
+=================
+Blitting Tutorial
+=================
+
+'Blitting' is a `standard technique
+<https://en.wikipedia.org/wiki/Bit_blit>`__ in computer graphics that
+in the context of matplotlib can be used to (drastically) improve
+performance of interactive figures.  It is used internally by the
+:mod:`~.animation` and :mod:`~.widgets` modules for this reason.
+
+The source of the performance gains is simply not re-doing work we do
+not have to.  For example, if the limits of an Axes have not changed,
+then there is no reason we should re-draw all of the ticks and
+tick-labels (particularly because text is one of the more expensive
+things to render).
+
+The procedure to save our work is roughly:
+
+- draw the figure, but exclude an artists marked as 'animated'
+- save a copy of the Agg RBGA buffer
+
+In the future, to update the 'animated' artists we
+
+- restore our copy of the RGBA buffer
+- redraw only the animated artists
+- show the resulting image on the screen
+
+thus saving us from having to re-draw everything which is _not_
+animated.
+
+Simple Example
+--------------
+
+We can implement this via methods on `.CanvasAgg` and setting
+``animated=True`` on our artist.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+x = np.linspace(0, 2*np.pi, 100)
+
+fig, ax = plt.subplots()
+# animated=True makes the artist be excluded from normal draw tree
+ln, = ax.plot(x, np.sin(x), animated=True)
+
+# stop to admire our empty window axes and ensure it is drawn
+plt.pause(.1)
+
+# save a copy of the image sans animated artist
+bg = fig.canvas.copy_from_bbox(fig.bbox)
+# draw the animated artist
+ax.draw_artist(ln)
+# show the result to the screen
+fig.canvas.blit(fig.bbox)
+
+for j in range(100):
+    # put the un-changed background back
+    fig.canvas.restore_region(bg)
+    # update the artist.
+    ln.set_ydata(np.sin(x + (j / 100) * np.pi))
+    # re-render the artist
+    ax.draw_artist(ln)
+    # copy the result to the screen
+    fig.canvas.blit(fig.bbox)
+
+
+###############################################################################
+# This example works and shows a simple animation, however because we are only
+# grabbing the background once, if the size of dpi of the figure change, the
```

Comment:
```suggestion
# The previous example works and shows a simple animation, however because we are only
# grabbing the background once, if the size or dpi of the figure change, the
```